src/gluonts/model/estimator.py

```diff
@@ -178,6 +186,18 @@ def train_model(self, training_data: Dataset) -> TrainOutput:
             float_type=self.float_type,
         )
 
+        validation_data_loader = (
+            InferenceDataLoader(
```

Comment:
I'm not sure whether the `InferenceDataLoader` does the right thing here. It runs the transformations in non-training mode, which does not provide the prediction target that is needed to calculate the loss. I'm surprised this doesn't throw an error, since the field should be missing.

Also the signature in the trainer is given as `Optional[TrainDataLoader]`.

Reply:
Fixed the signature and missing prediction target by using training transforms in `ValidationDataLoader`. It turns out that with the current state of transformations it is impossible to choose the non-training mode for the InstanceSplitter transformation — so, the `ValidationDataLoader` provides the way to estimate validation loss on a separate dataset that the user provides. I think that validation scheme that @lostella provided [here](https://github.com/awslabs/gluon-ts/pull/378#discussion_r337042900) depends on the algorithms — there is no common `prediction_length` field that can be used to handle all of the algorithms that GluonTS can use. 

So, the user is forced to provide a correct validation set in accordance with the algorithms and problems that they need to solve. 

cc @jaheba